azurerm/internal/services/datafactory/resource_arm_data_factory_linked_service_sftp.go

```diff
@@ -0,0 +1,288 @@
+		}
+	}
+
+	authenticationType := d.Get("authentication_type").(string)
+
+	host := d.Get("host").(string)
+	port := d.Get("port").(int)
+	username := d.Get("username").(string)
+	password := d.Get("password").(string)
+
+	passwordSecureString := datafactory.SecureString{
+		Value: &password,
+		Type:  datafactory.TypeSecureString,
+	}
+
+	sftpProperties := &datafactory.SftpServerLinkedServiceTypeProperties{
+		Host:               utils.String(host),
+		Port:               port,
+		AuthenticationType: datafactory.SftpAuthenticationType(authenticationType),
+		UserName:           utils.String(username),
+		Password:           &passwordSecureString,
+	}
+
+	description := d.Get("description").(string)
+
+	sftpLinkedService := &datafactory.SftpServerLinkedService{
+		Description:                           &description,
+		SftpServerLinkedServiceTypeProperties: sftpProperties,
+		Type:                                  datafactory.TypeSftp,
+	}
+
+	if v, ok := d.GetOk("parameters"); ok {
+		sftpLinkedService.Parameters = expandDataFactoryParameters(v.(map[string]interface{}))
+	}
+
+	if v, ok := d.GetOk("integration_runtime_name"); ok {
+		sftpLinkedService.ConnectVia = expandDataFactoryLinkedServiceIntegrationRuntime(v.(string))
+	}
+
+	if v, ok := d.GetOk("additional_properties"); ok {
+		sftpLinkedService.AdditionalProperties = v.(map[string]interface{})
+	}
+
+	if v, ok := d.GetOk("annotations"); ok {
+		annotations := v.([]interface{})
+		sftpLinkedService.Annotations = &annotations
+	}
+
+	linkedService := datafactory.LinkedServiceResource{
+		Properties: sftpLinkedService,
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, dataFactoryName, name, linkedService, ""); err != nil {
+		return fmt.Errorf("Error creating/updating Data Factory Linked Service SFTP Anonymous %q (Data Factory %q / Resource Group %q): %+v", name, dataFactoryName, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, dataFactoryName, name, "")
+	if err != nil {
+		return fmt.Errorf("Error retrieving Data Factory Linked Service SFTP Anonymous %q (Data Factory %q / Resource Group %q): %+v", name, dataFactoryName, resourceGroup, err)
+	}
+
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read Data Factory Linked Service SFTP Anonymous %q (Data Factory %q / Resource Group %q): %+v", name, dataFactoryName, resourceGroup, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmDataFactoryLinkedServiceSFTPRead(d, meta)
+}
+
+func resourceArmDataFactoryLinkedServiceSFTPRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DataFactory.LinkedServiceClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	dataFactoryName := id.Path["factories"]
+	name := id.Path["linkedservices"]
```

Comment:
And use a parse ID resource here, It looks like you could potentially share this parse ID function with all three resources